src/renderer/components/settings-execution.tsx

```diff
@@ -59,36 +101,201 @@ export class ExecutionSettings extends React.Component<ExecutionSettingsProps> {
   }
 
   /**
-   * Handles a change in the execution flags run with the Electron executable.
+   * Saves currently set execution flags or environment variables
+   * run with the Electron executable.
    *
-   * @param {React.ChangeEvent<HTMLInputElement>} event
+   * @param {SettingItemType} type
    */
-  public handleExecutionFlagChange(event: React.FormEvent<HTMLInputElement>) {
-    const { value } = event.currentTarget;
-    const flags = value.split('|');
-    this.props.appState.executionFlags = flags;
+  public handleSettingsItemSave(type: SettingItemType) {
+    const { executionFlags, environmentVariables } = this.state;
+    const { appState } = this.props;
+
+    if (type === SettingItemType.Flag) {
+      const flags = Object.values(executionFlags);
+      appState.executionFlags = flags.filter((f) => f !== '');
+    } else {
+      const vars = Object.values(environmentVariables);
+      appState.environmentVariables = vars.filter((v) => v !== '');
+    }
   }
 
   /**
-   * Handles a change in the environment variables run with the Electron executable.
+   * Handles a change in the execution flags or environment variables
+   * run with the Electron executable.
    *
    * @param {React.ChangeEvent<HTMLInputElement>} event
+   * @param {SettingItemType} type
    */
-  public handleEnvironmentVariableChange(
-    event: React.FormEvent<HTMLInputElement>,
+  public handleSettingsItemChange(
+    event: React.ChangeEvent<HTMLInputElement>,
+    type: SettingItemType,
   ) {
+    const { name, value } = event.currentTarget;
+
+    if (type === SettingItemType.Flag) {
+      this.setState((prevState) => ({
+        executionFlags: {
+          ...prevState.executionFlags,
+          [name]: value,
+        },
+      }));
+    } else {
+      this.setState((prevState) => ({
+        environmentVariables: {
+          ...prevState.environmentVariables,
+          [name]: value,
+        },
+      }));
+    }
```

Comment:
Nearly all the changes in settings-execution.tsx contain copypaste between handling flags and envvars. Would it make sense to extract out an abstract version of this and have both flags & envvars use that instead of repeating it?